src/clientdlg.cpp

```diff
@@ -277,7 +277,13 @@ CClientDlg::CClientDlg ( CClient*         pNCliP,
     {
         // initiate connection (always show the address in the mixer board
         // (no alias))
-        Connect ( strConnOnStartupAddress, strConnOnStartupAddress );
+
+        // initiate connection
+        // TODO: Refactor this for failing call on Connect()
+
+        pClient->Connect ( strConnOnStartupAddress, strConnOnStartupAddress );
+        // TODO: Find out why without this the mixer status issue still occurs. 
+        OnConnect ( strConnOnStartupAddress );
```

Comment:
Something called `OnAnything` should only be used as a connected signal handler, not called directly like this.

Reply:
See https://github.com/jamulussoftware/jamulus/pull/3372/files/0b6a2bb3c0c70329c85b48f0a6ae52c7e226c2f4#r1759507277

Reply:
Potentially, we could just emit pClient->Connect(...)?

Reply:
Probably the issue is that the connection happens in the constructor.
It would make sense to remove Connecting from the constructor and do it as in the other PR https://github.com/jamulussoftware/jamulus/pull/3364#discussion_r1759236825

Reply:
Ok. See https://forum.qt.io/topic/111772/signals-and-slots-within-same-class/3

Reply:
Yup: I said somewhere else, I think -- create the client, clientdlg and client json-rpc, wire them up, then tell the client to start running (passing in the start up details).  It would be in the "start running" that the network connection was initiated and signals start getting emitted from the network channel.

Reply:
You rejected connecting in main.cpp here: https://github.com/jamulussoftware/jamulus/pull/3364/files#r1759236825 ?

So how would you do it?

1. Create client
2. Create GUI
3. Create json rpc
4. Connect here by calling pClient->Connect() or what?

Reply:
See the changes.